src/catalog/src/remote/manager.rs

```diff
@@ -114,186 +104,65 @@ impl RemoteCatalogManager {
         }))
     }
 
-    async fn iter_remote_schemas(
-        &self,
-        catalog_name: &str,
-    ) -> Pin<Box<dyn Stream<Item = Result<SchemaKey>> + Send + '_>> {
-        let schema_prefix = build_schema_prefix(catalog_name);
-        let mut schemas = self.backend.range(schema_prefix.as_bytes());
-
-        Box::pin(stream!({
-            while let Some(r) = schemas.next().await {
-                let Kv(k, _) = r?;
-                if !k.starts_with(schema_prefix.as_bytes()) {
-                    debug!("Ignoring non-schema key: {}", String::from_utf8_lossy(&k));
-                    continue;
-                }
-
-                let schema_key = SchemaKey::parse(&String::from_utf8_lossy(&k))
-                    .context(InvalidCatalogValueSnafu)?;
-                yield Ok(schema_key)
-            }
-        }))
-    }
-
-    /// Iterate over all table entries on metasrv
-    async fn iter_remote_tables(
-        &self,
-        catalog_name: &str,
-        schema_name: &str,
-    ) -> Pin<Box<dyn Stream<Item = Result<(TableGlobalKey, TableGlobalValue)>> + Send + '_>> {
-        let table_prefix = build_table_global_prefix(catalog_name, schema_name);
-        let mut tables = self.backend.range(table_prefix.as_bytes());
-        Box::pin(stream!({
-            while let Some(r) = tables.next().await {
-                let Kv(k, v) = r?;
-                if !k.starts_with(table_prefix.as_bytes()) {
-                    debug!("Ignoring non-table prefix: {}", String::from_utf8_lossy(&k));
-                    continue;
-                }
-                let table_key = TableGlobalKey::parse(&String::from_utf8_lossy(&k))
-                    .context(InvalidCatalogValueSnafu)?;
-                let table_value =
-                    TableGlobalValue::from_bytes(&v).context(InvalidCatalogValueSnafu)?;
-
-                info!(
-                    "Found catalog table entry, key: {}, value: {:?}",
-                    table_key, table_value
-                );
-                // metasrv has allocated region ids to current datanode
-                if table_value
-                    .regions_id_map
-                    .get(&self.node_id)
-                    .map(|v| !v.is_empty())
-                    .unwrap_or(false)
-                {
-                    yield Ok((table_key, table_value))
-                }
-            }
-        }))
-    }
-
     /// Fetch catalogs/schemas/tables from remote catalog manager along with max table id allocated.
     async fn initiate_catalogs(&self) -> Result<HashMap<String, CatalogProviderRef>> {
         let mut res = HashMap::new();
 
         let mut catalogs = self.iter_remote_catalogs().await;
+        let mut joins = Vec::new();
```

Comment:
```suggestion
        let mut joins = Vec::with_capactity(catalogs.len());
```

Reply:
> Great job. Thanks a lot. But I found some CR comments were resolved without actual fixing. Please resolve the comments only when fixing them or leave a comment to explain why not do it.

Sorry, I fixed it locally, but haven't pushed yet. I will be careful afterwards.

Reply:
It seems that there is no `len` method in Stream.